### PR TITLE
Added a new function to allow printing help with additional arguments.

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@hardening @mfleisz @bmiklautz I've rebased this old pull. Anyone interested in merging?
It allows adding 'custom' command line arguments if FreeRDP is used as a library to the help print screen.

File: include/freerdp/client/cmdline.h [hunk 1]
```diff
@@ -28,21 +30,28 @@ extern "C" {
...
 
 FREERDP_API BOOL freerdp_parse_username(char* username, char** user, char** domain);
 FREERDP_API BOOL freerdp_parse_hostname(char* hostname, char** host, int* port);
 FREERDP_API BOOL freerdp_set_connection_type(rdpSettings* settings, int type);
 
 FREERDP_API BOOL freerdp_client_add_device_channel(rdpSettings* settings, int count, char** params);
 FREERDP_API BOOL freerdp_client_add_static_channel(rdpSettings* settings, int count, char** params);
-FREERDP_API BOOL freerdp_client_add_dynamic_channel(rdpSettings* settings, int count, char** params);
+FREERDP_API BOOL freerdp_client_add_dynamic_channel(rdpSettings* settings, int count,
+        char** params);
```
Comment (mfleisz) [inline include/freerdp/client/cmdline.h]: Do we really need a line break here?

File: winpr/libwinpr/utils/cmdline.c [hunk 2]
```diff
@@ -45,8 +45,9 @@
  *
  */
 
-int CommandLineParseArgumentsA(int argc, LPCSTR* argv, COMMAND_LINE_ARGUMENT_A* options, DWORD flags,
-		void* context, COMMAND_LINE_PRE_FILTER_FN_A preFilter, COMMAND_LINE_POST_FILTER_FN_A postFilter)
+int CommandLineParseArgumentsA(int argc, LPCSTR* argv, COMMAND_LINE_ARGUMENT_A* options,
+                               DWORD flags,
+                               void* context, COMMAND_LINE_PRE_FILTER_FN_A preFilter, COMMAND_LINE_POST_FILTER_FN_A postFilter)
```
Comment (mfleisz) [inline winpr/libwinpr/utils/cmdline.c]: Formatting looks strange (same for all other functions with these parameters)

**Comment by akallabeth** [comment]
@mfleisz Rebased again. If you like the pull, could you please merge?

**Pull request merged**